src/main/scala/mesosphere/marathon/core/task/Task.scala

```diff
@@ -309,6 +311,38 @@ object Task {
     override lazy val reservationId: String = instanceId.idString
   }
 
+  /**
+    * Identifier for app or pod task with and attempt count and type prefix.
+    *
+    * The ids match [[Task.TaskIdWithIncarnation.regex]] and include a launch attempt.
+    *
+    * Examples:
+    *  - "$tiwi$.myGroup_myApp.marathon-b6ff5fa5-7714-11e7-a55c-5ecf1c4671f6.$anon.1"
+    *  - "$tiwi$.myGroup_myApp.instance-b6ff5fa5-7714-11e7-a55c-5ecf1c4671f6.$anon.3"
+    *  - "$tiwi$.myGroup_myApp.marathon-b6ff5fa5-7714-11e7-a55c-5ecf1c4671f6.rails.2"
+    *  - "$tiwi$.myGroup_myApp.instance-b6ff5fa5-7714-11e7-a55c-5ecf1c4671f6.rails.42"
+    *
+    * @param instanceId Identifies the instance the task belongs to.
+    * @param containerName If set identifies the container in the pod. Defaults to [[Task.Id.Names.anonymousContainer]].
+    * @param attempt Counts how often a task has been launched on a specific reservation.
+    */
+  case class TaskIdWithIncarnation(val instanceId: Instance.Id, val containerName: Option[String], attempt: Long) extends Id {
```

Comment:
let's call this incarnation instead of attempt?

Reply:
Good point.